docs/react-18.md

```diff
@@ -0,0 +1,40 @@
+# React 18 & React Native
```

Comment:
Is this page still relevant? It is a year now that users on the New Architecture are using React 18... I feel like we can remove it. WDYT? 
cc. @cortinico @lunaleaps

Reply:
I think it still is as I recall it was updated recently. Can revisit this of course